seed/seed.go

```diff
@@ -160,26 +160,21 @@ type Seed interface {
 	Iter(f func(sn *Snap) error) error
 }
 
-// A SnapHandler can be used to perform any dedicated handling of seed
-// snaps and their digest computation while seed snap metadata loading
-// and verification is being performed.
-type SnapHandler interface {
-	// HandleAndDigestAssertedSnap should compute the digest of
-	// the given snap and perform any dedicated
-	// handling. essentialType is provided only for essential
-	// snaps.
-	// snapRev is provided by UC20+ seeds.
-	// deriveRev is provided by UC16/18 seeds, it can be used
-	// to get early access to the snap revision based on the digest.
-	// A different path can be returned if the snap has been copied
-	// elsewhere.
-	HandleAndDigestAssertedSnap(name, path string, essentialType snap.Type, snapRev *asserts.SnapRevision, deriveRev func(snapSHA3_384 string, snapSize uint64) (snap.Revision, error), tm timings.Measurer) (newPath, snapSHA3_384 string, snapSize uint64, err error)
-
-	// HandleUnassertedSnap should perfrom any dedicated handling
-	// for the given unasserted snap.
-	// A different path can be returned if the snap has been copied
-	// elsewhere.
-	HandleUnassertedSnap(name, path string, tm timings.Measurer) (newPath string, err error)
+// A ContainerHandler can be used to perform any dedicated handling of seed
+// snaps/components and their digest computation while seed snap/component
+// metadata loading and verification is being performed.
+type ContainerHandler interface {
+	// HandleAndDigestAssertedContainer should compute the digest of the
+	// given container and perform any dedicated handling. A different path
+	// can be returned if the container has been copied elsewhere.
+	HandleAndDigestAssertedContainer(cpi snap.ContainerPlaceInfo, path string,
+		tm timings.Measurer) (newPath, snapSHA3_384 string, snapSize uint64, err error)
+
+	// HandleUnassertedContainer should perform any dedicated handling for
+	// the given unasserted snap/component. A different path can be
+	// returned if the container has been copied elsewhere.
```

Comment:
please leave a TODO about adding back extra information like the type to support optimizations,